packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -650,21 +637,18 @@ export class SDKClient {
 
       throw sdkClientError;
     } finally {
-      /**
-       * @note Capturing the charged transaction fees at the end of the flow ensures these fees are eventually
-       *       captured in the metrics and rate limiter class, even if SDK transactions fail at any point.
-       */
+      // emitting an EXECUTE_QUERY event to kick off capturing metrics process asynchronously
       if (queryCost && queryCost !== 0) {
-        this.addExpenseAndCaptureMetrics(
-          `TransactionExecution`,
-          query.paymentTransactionId!,
-          queryType,
+        this.eventEmitter.emit(constants.EVENTS.EXECUTE_QUERY, {
+          executionType: `TransactionExecution`,
+          transactionId: query.paymentTransactionId?.toString()!,
```

Comment:
Since `IExecuteQueryEventPayload` is also reused in `captureTransactionMetrics`, which does not have `query.paymentTransactionId`, I'd keep it as `transactionId` to maintain its generic nature.